fastapi/encoders.py

```diff
@@ -142,6 +105,54 @@ def jsonable_encoder(
             )
         return encoded_list
 
+    def encode_BaseModel(obj: BaseModel) -> Any:
+        encoder = getattr(obj.__config__, "json_encoders", {})
+        if custom_encoder:
+            encoder.update(custom_encoder)
+
+        obj_dict = obj.dict(
+            include=include,
+            exclude=exclude,
+            by_alias=by_alias,
+            exclude_unset=exclude_unset,
+            exclude_none=exclude_none,
+            exclude_defaults=exclude_defaults,
+        )
+        if "__root__" in obj_dict:
+            obj_dict = obj_dict["__root__"]
+
+        return jsonable_encoder(
+            obj_dict,
+            exclude_none=exclude_none,
+            exclude_defaults=exclude_defaults,
+            custom_encoder=encoder,
+            sqlalchemy_safe=sqlalchemy_safe,
+        )
+
+    # Use type comparsions on common types before expensive isinstance checks
+    if type(obj) in (str, int, float, type(None)):
+        return obj
+    if type(obj) == dict:
+        return encode_dict(obj)
+    if type(obj) in (list, set, frozenset, GeneratorType, tuple):
+        return encode_array(obj)
+
+    if isinstance(obj, (str, int, float, type(None))):
+        return obj
+    if isinstance(obj, dict):
+        return encode_dict(obj)
+    if isinstance(obj, (list, set, frozenset, GeneratorType, tuple)):
+        return encode_array(obj)
+    if isinstance(obj, BaseModel):
```

Comment:
Maybe this case should go before all the other `isinstance` checks, as it's quite common to have your endpoint return pydantic models.

Maybe we could move to the bottom all the instance checks for derived types of str, int, etc, as those cases should be rather rare.

How was the order of the checks defined?

Reply:
The order was kinda defined arbitrarily based on what I thought would be most common. I also originally thought that the `BaseModel`'s Fields would be counted as `str` instances, however since `BaseModel`s are getting converted to a `dict` they'll always be `str`. I've re-arranged the isinstnace checks to be more logical.